### ISSUE TITLE
ATI Driver doesn't seem to have access to RAM (shown in big picture mode)

### ISSUE THREAD

**Issue by boombatower** [ValveSoftware/steam-for-linux#1311]
Assume this is ATI driver issue so I will cross-post in their forum, but the big picture mode doesn't seem to think I have much ram when the card has 6GB (7970 Ghz 6GB).

![image](https://f.cloud.github.com/assets/209374/131099/f0d51a46-7048-11e2-88dd-baea06d9b614.png)

```
Processor Information:
    Vendor:  GenuineIntel
    Speed: 3501 Mhz
    8 logical processors
    4 physical processors
    HyperThreading:  Supported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Unsupported
    SSE41:  Supported
    SSE42:  Supported

Network Information:
    Network Speed:  

Operating System Version:
    "openSUSE 12.2 (x86_64)" (64 bit)
    Kernel Name:  Linux
    Kernel Version:  3.7.6-22-desktop
    X Server vendor:  The X.Org Foundation
    X Server release:  11302000

Video Card:
    Driver:  ATI Technologies Inc. AMD Radeon HD 7900 Series 

    Driver Version:  4.2.12002 Compatibility Profile Context 9.012
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 59 Hz
    VendorID Not Detected
    DeviceID Not Detected
    Number of Monitors:  3
    Number of Video Cards Not Detected
    Primary Display Resolution:  2560 x 1600
    Desktop Resolution: 6400 x 1600
    Primary Display Size: 25.24" x 15.75"  (29.72" diag)
                                            64.1cm x 40.0cm  (75.5cm diag)
    Primary VRAM Not Detected

Sound card:
    Audio device: Realtek ALC892

Memory:
    RAM:  32190 Mb

Miscellaneous:
    UI Language:  English
    LANG:  en_US.UTF-8
    Microphone:  Not set
    Total Hard Disk Space Available:  202061 Mb
    Largest Free Hard Disk Block:  189114 Mb
```

**Comment by vpltd-peterm** [comment]
I will note that neither the nvidia binary driver nor fglrx implement GLX_MESA_query_renderer support. This is unfortunate, as it would be most useful.

Using lspci for VRAM detection hasn't worked for a long time. Graphics cards only map portions of VRAM through an aperture, and the only thing lspci will tell you is the size of this aperture.

**Comment by baos780** [comment]
NVIDIA VRAM DETECTED!  ubuntu with asus 960 card.
my r9 270's are going in the garbage! lol


Processor Information:
    Vendor:  AuthenticAMD
    CPU Family:  0x15
    CPU Model:  0x2
    CPU Stepping:  0x0
    CPU Type:  0x0
    Speed:  4000 Mhz
    8 logical processors
    8 physical processors
    HyperThreading:  Unsupported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Supported
    SSE41:  Supported
    SSE42:  Supported
    
Network Information:
    Network Speed:  
    
Operating System Version:
    Ubuntu 14.04.2 LTS (64 bit)
    Kernel Name:  Linux
    Kernel Version:  3.13.0-46-generic
    X Server Vendor:  The X.Org Foundation
    X Server Release:  11501000
    X Window Manager:  GNOME Shell
    Steam Runtime Version:  steam-runtime-release_2015-01-06
    
Video Card:
    Driver:  NVIDIA Corporation GeForce GTX 960/PCIe/SSE2

    Driver Version:  4.5.0 NVIDIA 346.35
    OpenGL Version: 4.5
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 60 Hz
    VendorID:  0x10de
    DeviceID:  0x1401
    Number of Monitors:  1
    Number of Logical Video Cards:  1
    Primary Display Resolution:  1920 x 1080
    Desktop Resolution: 1920 x 1080
    Primary Display Size: 73.70" x 41.46"  (84.53" diag)
                                            187.2cm x 105.3cm  (214.7cm diag)
    Primary Bus: PCI Express 16x
    Primary VRAM: 2048 MB
    Supported MSAA Modes:  2x 4x 8x 16x 
    
Sound card:
    Audio device: AV200
    
Memory:
    RAM:  16006 Mb
    
Miscellaneous:
    UI Language:  English
    LANG:  en_CA.UTF-8
    Microphone:  Not set
    Total Hard Disk Space Available:  17705 Mb
    Largest Free Hard Disk Block:  9998 Mb
    
Installed software:
    
Recent Failure Reports:

**Comment by boombatower** [comment]
Still occurs on recent Mesa for ATI card (radeon hd 7970 ghz edition). Big Picture functions excellently though. :)

```
Video Card:
    Driver:  X.Org Gallium 0.4 on AMD TAHITI
    Driver Version:  3.0 Mesa 10.5.3
    OpenGL Version: 3.0
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 59 Hz
    VendorID Not Detected
    DeviceID Not Detected
    Number of Monitors:  3
    Number of Video Cards Not Detected
    Primary Display Resolution:  2560 x 1600
    Desktop Resolution: 6400 x 1600
    Primary Display Size: 25.24" x 15.75"  (29.72" diag)
                                            64.1cm x 40.0cm  (75.5cm diag)
    Primary VRAM Not Detected
```

As per [above comment](/ValveSoftware/steam-for-linux/issues/1311#issuecomment-51694037) I checked for the availability of `GLX_MESA_query_renderer` extension and it is present.

```
$ glxinfo 
name of display: :0
display: :0  screen: 0
direct rendering: Yes
server glx vendor string: SGI
server glx version string: 1.4
server glx extensions:
    GLX_ARB_create_context, GLX_ARB_create_context_profile, 
    GLX_ARB_fbconfig_float, GLX_ARB_framebuffer_sRGB, GLX_ARB_multisample, 
    GLX_EXT_create_context_es2_profile, GLX_EXT_framebuffer_sRGB, 
    GLX_EXT_import_context, GLX_EXT_texture_from_pixmap, GLX_EXT_visual_info, 
    GLX_EXT_visual_rating, GLX_INTEL_swap_event, GLX_MESA_copy_sub_buffer, 
    GLX_OML_swap_method, GLX_SGIS_multisample, GLX_SGIX_fbconfig, 
    GLX_SGIX_pbuffer, GLX_SGIX_visual_select_group, GLX_SGI_swap_control
client glx vendor string: Mesa Project and SGI
client glx version string: 1.4
client glx extensions:
    GLX_ARB_create_context, GLX_ARB_create_context_profile, 
    GLX_ARB_create_context_robustness, GLX_ARB_fbconfig_float, 
    GLX_ARB_framebuffer_sRGB, GLX_ARB_get_proc_address, GLX_ARB_multisample, 
    GLX_EXT_buffer_age, GLX_EXT_create_context_es2_profile, 
    GLX_EXT_fbconfig_packed_float, GLX_EXT_framebuffer_sRGB, 
    GLX_EXT_import_context, GLX_EXT_texture_from_pixmap, GLX_EXT_visual_info, 
    GLX_EXT_visual_rating, GLX_INTEL_swap_event, GLX_MESA_copy_sub_buffer, 
    GLX_MESA_multithread_makecurrent, GLX_MESA_query_renderer, 
    GLX_MESA_swap_control, GLX_OML_swap_method, GLX_OML_sync_control, 
    GLX_SGIS_multisample, GLX_SGIX_fbconfig, GLX_SGIX_pbuffer, 
    GLX_SGIX_visual_select_group, GLX_SGI_make_current_read, 
    GLX_SGI_swap_control, GLX_SGI_video_sync
GLX version: 1.4
GLX extensions:
    GLX_ARB_create_context, GLX_ARB_create_context_profile, 
    GLX_ARB_fbconfig_float, GLX_ARB_framebuffer_sRGB, 
    GLX_ARB_get_proc_address, GLX_ARB_multisample, 
    GLX_EXT_create_context_es2_profile, GLX_EXT_framebuffer_sRGB, 
    GLX_EXT_import_context, GLX_EXT_texture_from_pixmap, GLX_EXT_visual_info, 
    GLX_EXT_visual_rating, GLX_INTEL_swap_event, GLX_MESA_copy_sub_buffer, 
    GLX_MESA_multithread_makecurrent, GLX_MESA_query_renderer, 
    GLX_MESA_swap_control, GLX_OML_swap_method, GLX_OML_sync_control, 
    GLX_SGIS_multisample, GLX_SGIX_fbconfig, GLX_SGIX_pbuffer, 
    GLX_SGIX_visual_select_group, GLX_SGI_make_current_read,                                                                         
    GLX_SGI_swap_control, GLX_SGI_video_sync                
....
```

Nothing new I suppose.

**Comment by Fabioamd87** [comment]
same error here, 268.44Mb on Radeon 5450 with 512Mb

**Comment by Tele42** [comment]
As noted elsewhere, this appears to be strictly a cosmetic issue, this message does not affect the amount of video memory available to games or reduce performance.

**Comment by Fabioamd87** [comment]
thanks

**Comment by baos780** [comment]
That was not my experience.  I had two ati video cards that should have overpowered the nvidia 970 I replaced them with.  Instead I now have smooth graphics at any framerate.

**Comment by boombatower** [comment]
A lot more variables than just the VRAM. I an see how much VRAM games / steam uses and it's more than it detects. Still given people may have that impressions...these are the polish things that make people feel like they are using something solid.

**Comment by DanielPower** [comment]
I'm getting the same message on a HD 6950 with Catalyst 15.302 on Linux Mint 17.3
The card shows 1024mb vram in Catalyst Control Center, but Steam detects only 260

**Comment by jyrkimanboy** [comment]
I experience this too, and it does have an effect on performance.
I have amd r9 390 wth 8Gb, and steam detects 256Mb.
The rest: amd fx8350, 16Gb ram, asus crosshair v formula z.
Linux mint 17.3 cinnamon, driver fglrx 15.302.
I'm (maybe because of this) unable to set the resolution to my native 1440p,
and the game then runs at a lower resolution, plus the steam big picture is partially outside of my screen, unaccessible.
The game, cities-skylines, runs at 14 fps. This is too low for the hardware i have.

**Comment by DePingus** [comment]
Did we ever get a fix for this? I get this warning in Big Picture and I get very noticeable input lag just navigating the Big Picture menus using the Steam Link to stream from a Linux host. This host is running Fedora 23 with an AMD R9 270 on Mesa 11.1.0. If I boot  the same machine into Windows10 the lag goes away.

**Comment by stevethepocket** [comment]
I still get the message, even with my new card that the System Information dialog correctly identifies as having 2GB (in Windows it says I have -2GB, which is amusing), but performance doesn't seem laggy at all so I have no idea what's up.

**Comment by DanielPower** [comment]
I'm also getting the message on my R9 270X, regardless of what driver I use. I've used various versions of Catalyst fglrx, Mesa 11.2, and Mesa 11.3-git.

No performance issues, but some games prevent me from choosing high texture settings, because they think I don't have enough VRAM.

**Comment by boombatower** [comment]
An interest note, while the issue still persists in big picture the steam system information shows the correct value:
```
 Primary VRAM: 6144 MB
```

Perhaps copy/paste or share code? :-)

For kicks, much update software:
```
Distro: openSUSE Tumbleweed [20160913]
Kernel version: 4.7.3-1-default
OpenGL version string: 3.0 Mesa 12.0.2
OpenGL renderer string: Gallium 0.4 on AMD TAHITI (DRM 2.45.0 / 4.7.3-1-default, LLVM 3.8.0)
OpenGL core profile version string: 4.1 (Core Profile) Mesa 12.0.2
Xorg version : 7.6_1.18.4-1.2
xf86-video-ati version : 7.7.0-1.3
DRI3 enabled
Hardware: Radeon HD 7970 ghz
```

**Comment by kisak-valve** [comment]
Related: https://lists.freedesktop.org/archives/mesa-dev/2017-January/142635.html

**Comment by kisak-valve** [comment]
This issue should now be resolved.

First, in the kernel this was resolved in [linux 3.17](https://cgit.freedesktop.org/mesa/mesa/commit/?id=31136eae3ae03615a244496d4f82be50e82d803c), and now in [mesa git](https://cgit.freedesktop.org/mesa/mesa/commit/?id=31136eae3ae03615a244496d4f82be50e82d803c). Waiting for confirmation in a point release of mesa before closing.

Note: This issue should be strictly cosmetic and general performance is not expected to significantly change.

**Comment by kisak-valve** [comment]
Closing as this should work with mesa with any kernel 3.17 or newer, fglrx is deprecated, and there are no reports of an issue with amdgpu.
